File: src/main/java/com/manicure/service/CartService.java
package com.manicure.service;

import java.util.List;

import entity.Cart;

public interface CartService {

	/**
	 * 添加到购物车
	 * @param cartList
	 * @param itemId
	 * @param num
	 * @return
	 */
	public List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num);
	
	/**
	 * 在 redis 中查询购物车
	 * @param username
	 * @return
	 */
	public List<Cart> findCartListFromRedis(String username);
	
	/**
	 * 将购物车保存到 redis
	 * @param username
	 * @param cartList
	 */
	public void saveCartListToRedis(String username,List<Cart> cartList);

	/**
	 * 合并购物车
	 * @param cartList1
	 * @param cartList2
	 * @return
	 */
	public List<Cart> mergeCartList(List<Cart> cartList1,List<Cart> cartList2);
	
	/**
	 * 设置是否选中
	 * @param cartList
	 * @param itemId
	 * @param isSelected
	 * @return
	 */
	public List<Cart> updateSelection(List<Cart> cartList, Long itemId, String isSelected);
}

File: src/main/webapp/js/service/cartService.js
//服务层
app.service('cartService',function($http){
	    	
	
	
	//获取购物列表
	this.findCartList=function(){
		return $http.get('../cart/findCartList.do');
	}    
	
	//添加到购物车
	this.addGoodsToCartList=function(itemId,num){
		return $http.get('../cart/addGoodsToCartList.do?itemId='+itemId+"&num="+num);
	} 
	
	//求合计数
	this.total=function(cartList){
		var totalValue={totalNum:0,totalMoney:0,cartListSelected:[]};
			
		for(var i=0;i<cartList.length ;i++){
			var cart=cartList[i];//购物车对象
			if(cart.seleted==true){
				var orderItem=  cart.orderItem;
				totalValue.cartListSelected.push(orderItem);
				totalValue.totalNum+=orderItem.num;//累加数量
				totalValue.totalMoney+=orderItem.totalFee;//累加金额	
			}
		}
		return totalValue;
	}
//	cartListSelected

	//是否已选
	this.updateSelection=function(itemId,isSelected){
		return $http.get('../cart/updateSelection.do?itemId='+itemId+"&isSelected="+isSelected);
	} 
	
	//是否已选
	this.findCartListSelected=function(){
		return $http.get('../cart/findCartListSelected.do');
	} 
	
	//保存订单
	this.submitOrder=function(order){
		return $http.post('../order/add.do',order);		
	}
		
});
